extend-string.js

```diff
@@ -0,0 +1,126 @@
+/* Checks for vowel(s) in a string */
+String.prototype.hasVowels = function() {
+  let regEx = /[aeiou]/g; // pattern to search for vowels 
+  return regEx.test(this)? true : false
+}
+
+/* Converts each string character to uppercase */
+String.prototype.toUpper = function() {
+  let result = ''; // empty string variable to store result
+  for(index = 0; index < this.length; index++){ 
+    let characterCode = this.charCodeAt(index); // get the character code of each character
+    if(characterCode >= 97 && characterCode <= 122){ //check if character code is in lowercase using ascii code
+      characterCode -= 32 // subtract 32 from the character code to get its uppercase equivalent code
+    } 
+    result += String.fromCharCode(characterCode); //appending each of the transformed character to the result variable
+  }
+  return result // return the 
+}
+
+/* Converts each string character to lowercase */
+String.prototype.toLower = function() {
+  let result = ''; // empty string variable to store result
+  for(index = 0; index < this.length; index++){ 
+    let characterCode = this.charCodeAt(index); // get the character code of each character
+    if(characterCode >= 65 && characterCode <= 90){ //check if character code is in uppercase using ascii code
+      characterCode += 32 // add 32 to the character code to get its lowercase equivalent code
+    } 
+    result += String.fromCharCode(characterCode); //appending each of the transformed character to the result variable
+  }
+  return result
+};
+
+/* Converts first letter of a string to uppercase */
+String.prototype.ucFirst = function() {
+  if(/[A-Z]/.test(this)) return 'First character already capitalized'
```

Comment:
I suggest this line be removed as it will only cause your users more checks before they can use the result. Try testing this again sEUn. 
```suggestion
  if(/^[A-Z]{1}.*/.test(this)) return 'First character already capitalized'
```